lib/models/device.coffee

```diff
@@ -1117,13 +1117,16 @@ exports.hasDeviceUrl = (uuid, callback) ->
 # });
 ###
 exports.getDeviceUrl = (uuid, callback) ->
-	exports.hasDeviceUrl(uuid).then (hasDeviceUrl) ->
+	exports.hasDeviceUrl(uuid).bind({}).then (hasDeviceUrl) ->
 		if not hasDeviceUrl
 			throw new Error("Device is not web accessible: #{uuid}")
 
 		return configModel.getAll().get('deviceUrlsBase')
 	.then (deviceUrlsBase) ->
-		return "https://#{uuid}.#{deviceUrlsBase}"
+		this.deviceUrlsBase = deviceUrlsBase
```

Comment:
Awesome thanks! I've just read so many things about nested promises that I try an avoid them completely, but one level down is fine :) Cheers!